gdc_client/query/versions.py

```diff
@@ -29,6 +29,17 @@ def get_latest_versions(url, uuids, verify=True):
     for chunk in _chunk_list(uuids):
         resp = requests.post(versions_url, json={"ids": chunk}, verify=verify)
 
+        # anything greater than or equal to a 400 status code is not allowed
+        if not resp.ok:
+            raise Exception(
```

Comment:
This reads much better. The only nit picky things I have are A) could we use a more precise error? Either built-in or custom. And B) is the above comment needed? 

I understand the fear around not getting back a valid json. However, if we got back a invalid json from the service for an ok response where we requested json, that would mean the service was completely broken and I don't see that as likely. Plus, we'd prob know because we'd be having failures in a lot more places than this. The 502 isn't a valid json because that level of response isn't necessarily coming from the service itself at all.

Reply:
it seems like the norm for dtt to just throw generic Exception for every error. I added the comment because I didn't know that resp.ok is True for both 2XX and 3xx errors (I would've just thought 2XX errors). Could remove it though if its not needed

Reply:
Ok is a pretty universal 400 or less because 300s tech are errors they are redirects. If you are worried about 300s tho, you can always do an explicit check of the status code to make sure its in the 200's instead of using the ok prop. 

As for the more applicable exception, this shouldn't cause any problem at all. It just makes the expected behavior of this code more explicit. Also you should make note of the exception that is raised in the doc string.